cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java

```diff
@@ -898,35 +899,47 @@ private boolean checkNodeConfig(StartUpStatus remoteStartUpStatus, AddNodeRespon
    * Check if the seed nodes are consistent with other nodes. Only used when establishing the
    * initial cluster.
    */
-  private void checkSeedNodes() {
+  private void checkSeedNodesStatus()
+      throws ConfigInconsistentException, StartUpCheckFailureException {
     boolean canEstablishCluster = false;
     long startTime = System.currentTimeMillis();
     AtomicInteger consistentNum = new AtomicInteger(1);
     AtomicInteger inconsistentNum = new AtomicInteger(0);
     while (!canEstablishCluster) {
       consistentNum.set(1);
       inconsistentNum.set(0);
-      checkSeedNodesOnce(consistentNum, inconsistentNum);
-
+      checkSeedNodesStatusOnce(consistentNum, inconsistentNum);
       canEstablishCluster = analyseStartUpCheckResult(consistentNum.get(), inconsistentNum.get(),
-          getAllNodes().size(), System.currentTimeMillis() - startTime);
+          getAllNodes().size());
+      // If reach the start up time threshold, shut down.
+      // Otherwise, wait for a while, start the loop again.
+      if (System.currentTimeMillis() - startTime > ClusterUtils.START_UP_TIME_THRESHOLD_MS) {
+        throw new StartUpCheckFailureException();
+      } else {
+        try {
+          Thread.sleep(ClusterUtils.START_UP_CHECK_TIME_INTERVAL_MS);
+        } catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
+          logger.error("Unexpected interruption when waiting for next start up check", e);
+        }
+      }
     }
   }
 
-  private void checkSeedNodesOnce(AtomicInteger consistentNum, AtomicInteger inconsistentNum) {
-    ExecutorService pool = new ScheduledThreadPoolExecutor(STARTUP_CHECK_THREAD_POOL_SIZE);
+  private void checkSeedNodesStatusOnce(AtomicInteger consistentNum,
+      AtomicInteger inconsistentNum) {
+    ExecutorService pool = new ScheduledThreadPoolExecutor(getAllNodes().size());
```

Comment:
If user uses `sh sbin/start-server.sh` to start server, every node in seednodes should be checked, thus the minimum should be  'getAllNodes().size() - 1'.
If user use `sh sbin/add-server.sh` to add server,then this funcation will never be called,the newly added server will try to check Status by the node which will help it to join the cluster.
So it seems no problem will occur?

Reply:
Yes. But I'm not sure if it is possible that there are tens of seed nodes. And if in that case, will the core pool size be too large for the machine?

Reply:
I think it's OK to create dozens of threads to accelerate the start-up check as it will only be used during building cluster, so it will cause on effect to cluster's read or write performance. As for the larger cluster such as hundreds of nodes, I think we should change not only the current design of the meta group but also the current p2p mode of checkStatus.
So maybe it's almost OK for current design.